### ISSUE TITLE
logits or log probabilities?

### ISSUE THREAD

**Issue by jacksonloper** [tensorflow/tensorflow#17177]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.4 and 1.5
- **Python version**:  3.6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: Tried several
- **GPU model and memory**: CPU and GPU, but GPU was nvidia p100
- **Exact command to reproduce**: See "source code" below

### Describe the problem
There seems to be some confusion about logits vs log probabilities.  In some cases logits seem to mean just log probabilities.  In other cases they seem to mean actual logits, `log(p/(1-p))` which is quite a different thing.  It seems possible to me that this is causing downstream bugs, potentially big ones in edward, though I haven't looked closely.  

Other people [also seem confused](https://stackoverflow.com/questions/41455101/what-is-the-meaning-of-the-word-logits-in-tensorflow).

### Source code / logs
```
import tensorflow as tf
from tensorflow.python.ops.distributions import util as distribution_util
import numpy as np

tf.reset_default_graph()
sess=tf.Session()

inp = tf.placeholder(dtype=tf.float32)
logits,probs=distribution_util.get_logits_and_probs(logits=inp)

categ=tf.distributions.Categorical(logits=inp)
vals=categ.sample(100000)

probsv,valsv=sess.run([probs,vals],feed_dict={inp:np.log([.2,.3,.5])})

print(probsv)
print([np.mean(valsv==x) for x in range(3)])
```

```
[0.16666667 0.23076922 0.33333334]
[0.20078, 0.29984, 0.49938]
```

So in some cases `logits` seems to mean log probabilities, but on other cases
it seem to actually mean logits.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by srvasude** [comment]
Hi!

`get_logits_and_probs` takes an argument `multidimensional`. which specifies whether to compute `log(p)` vs `log(p/(1-p))`.

get_logits_and_probs is also called internally by distribution objects, so there isn't a need to call it before passing in arguments (the name isn't exported into the tf namespace for that reason).

That being said, I do agree the name is confusing (the function was written with logits in mind, but was reworked to also handle log(prob) in certain cases) , and am happy to iterate on getting a better name for this. There is also potential to clean up some of the names for distributions (e.g. Categorical accepts `logits`, but these are really log probs).

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jart** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
